cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -219,6 +249,50 @@ private void populateResponse(IThrowableProxy throwableProxy,
     }
   }
 
+  @Nullable
+  private ErrorClassificationResponse findMatchingRule(CloseableIterator<LogEvent> logIter) {
+    while (logIter.hasNext()) {
+      ILoggingEvent logEvent = logIter.next().getLoggingEvent();
+      IThrowableProxy throwableProxy = logEvent.getThrowableProxy();
+      while (throwableProxy != null) {
+        for (ErrorClassificationRule rule : ruleList) {
+          if (throwableProxy.getClassName().equals(rule.getExceptionClassName())
+              && codeMethodMatched(rule.getCodeMethodRegex(),
+              throwableProxy.getStackTraceElementProxyArray())) {
+            LOG.debug("Rule {} matched", rule);
+            String errorMessage = throwableProxy.getMessage();
+            ErrorCategory errorCategory = new ErrorCategory(ErrorCategoryEnum.OTHERS, rule.getId());
+            return new ErrorClassificationResponse.Builder()
+                .setErrorCategory(errorCategory.getErrorCategory())
```

Comment:
This look strange. You construct an object and immediately get ErrorCategoryEnum.OTHERS from it. Or is it going to return anything else? BTW: Why is it always OTHER?

Reply:
we also add `ruleId` as `subCategory` in it.

Reply:
On a second thought, I removed ruleId from `errorCategory` and added it only in the metric so that it is not visible to users directly.

Reply:
I would still allow it to be specified and not OTHER. Or check MDC tags if we can figure something out automatically. Making it always OTHER limits value of the rules